test/gtest/mem_buffer.h

```diff
@@ -0,0 +1,203 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MEM_BUFFER_H
+#define MEM_BUFFER_H
+
+#include "nixl_types.h"
+#include <cstdint>
+#include <stdexcept>
+#include <vector>
+#include <random>
+#include <memory>
+#include <type_traits>
+
+namespace gtest {
+
+template<nixl_mem_t MemType> class MemBuffer;
```

Comment:
since gtest is not data path, i think it would simplify the code to make mem_type a run time parameter and not a template argument.
this way we can iterate on all mem type combinations using a simple loop.
imagine having even more memory types (OBJ, FILE, ...) - having this as template would exponentially increase the code size.

Reply:
The goal of templatizing MemBuffer was not to improve runtime performance by removing run-time conditionals, but to improve code readability and maintainability by having a clean separate implementations of MemBuffer for every memory type instead of a single class with "if (DRAM) .. else if (VRAM) .. else if ..." in every one of its methods. Having some boilerplate in TEST_P blocks looks like a lesser evil to me.